starter-core/src/test/groovy/io/micronaut/starter/feature/buildless/BuildlessSpec.groovy

```diff
@@ -0,0 +1,68 @@
+package io.micronaut.starter.feature.buildless
+
+import io.micronaut.starter.ApplicationContextSpec
+import io.micronaut.starter.BuildBuilder
+import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.application.Project
+import io.micronaut.starter.build.gradle.GradleBuild
+import io.micronaut.starter.build.gradle.GradleDsl
+import io.micronaut.starter.feature.build.MicronautGradleEnterprise
+import io.micronaut.starter.feature.build.gradle.templates.settingsGradle
+import io.micronaut.starter.fixture.CommandOutputFixture
+import io.micronaut.starter.options.BuildTool
+import io.micronaut.starter.options.Language
+import spock.lang.Subject
+
+class BuildlessSpec extends ApplicationContextSpec implements CommandOutputFixture {
```

Comment:
[See my comment](https://github.com/micronaut-projects/micronaut-starter/pull/2171#issuecomment-1806191525) about how we can change this to check the settings file and pick up the version from the template.version.toml file

I tried to push, but can't make changes to this PR, but this is what I ended up with for a test here

```groovy
class BuildlessSpec extends ApplicationContextSpec implements CommandOutputFixture {

    void 'test README.md with buildless feature contains links to docs'() {
        when:
        def output = generate([Buildless.NAME])
        def readme = output["README.md"]

        then:
        readme.contains("[https://docs.less.build/](https://docs.less.build/)")
    }

    void "feature buildless properties validation"() {
        given:
        def feature = beanContext.getBean(Buildless)

        expect:
        for (applicationType in ApplicationType.values()) {
            assert feature.supports(applicationType)
        }

        feature.category == Category.DEV_TOOLS
        feature.name == Buildless.NAME
    }

    void "buildless configured correctly for #buildTool"() {
        when:
        def output = generate(ApplicationType.DEFAULT, new Options(Language.JAVA, buildTool), [Buildless.NAME])
        def version = beanContext.getBean(CoordinateResolver).resolve(Buildless.BUILDLESS_PLUGIN_ARTIFACT).get().version
        def settings = buildTool == BuildTool.GRADLE ? output['settings.gradle'] : output['settings.gradle.kts']
        def expectedPlugin = buildTool == BuildTool.GRADLE ? """id "build.less" version "$version\"""" : """id("build.less") version("$version")"""

        then: 'we have the plugin in the settings file'
        settings.contains("    $expectedPlugin")

        and: 'we have the configuration in the settings file'
        settings.contains("""\
buildless {
    // this call is enough to configure the cache, although, remember to
    // set BUILDLESS_APIKEY in your environment.
}""")

        and: 'caching is enabled in the properties file'
        output['gradle.properties'].contains("org.gradle.caching=true")

        where:
        buildTool << BuildTool.valuesGradle()
    }

    void "io.micronaut.starter.feature.buildless.Buildless is visible"() {
        expect:
        beanContext.getBean(Buildless).visible
    }
}
```